pkgs/applications/misc/viking/default.nix

```
@@ -1,24 +1,39 @@
 { fetchurl, stdenv, makeWrapper, pkgconfig, intltool, gettext, gtk2, expat, curl
 , gpsd, bc, file, gnome-doc-utils, libexif, libxml2, libxslt, scrollkeeper
-, docbook_xml_dtd_412, gexiv2, sqlite, gpsbabel, expect
-, geoclue2, liboauth, nettle }:
+, docbook_xml_dtd_412, gexiv2, gpsbabel, expect
+, withMapnik ? false, mapnik
+, withMBTiles ? true, sqlite
+, withOAuth ? true, liboauth
+, withMd5Hash ? true, nettle
+, withGeoClue ? true, geoclue2 }:
 
 stdenv.mkDerivation rec {
   pname = "viking";
-  version = "1.7";
+  version = "1.8";
 
   src = fetchurl {
     url = "mirror://sourceforge/viking/viking/viking-${version}.tar.bz2";
-    sha256 = "092q2dv0rcz12nh2js1z1ralib1553dmzy9pdrvz9nv2vf61wybw";
+    sha256 = "1a0g0fbj4q5s9p8fv0mqvxws10q3naj81l72sz30vvqpbz6vqp45";
   };
 
+  patches = [ ./0001-fix-build.patch ];
```

please use `fetchpatch` here:
```suggestion
  patches = [
    # Fix build without mapnik and sqlite https://github.com/viking-gps/viking/pull/79
    (fetchpatch {
      url = "https://github.com/viking-gps/viking/commit/995feefcb97bdb1590ed018224cf47ce197fe0c1.patch";
      sha256 = "1xb0b76kg690fag9mw3yfj5k766jmqp1sm8q4f29n1h3nz5g8izd";
    })
  ];
```

---

Fixed.